components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/AuthenticationHandler.java

```diff
@@ -127,9 +128,18 @@ protected void postAuthenticate(MessageContext messageContext, AuthenticationRes
                     String organizationId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
                     if (StringUtils.isNotBlank(organizationId)) {
                         try {
-                            Optional<org.wso2.carbon.user.core.common.User> resolvedUser = AuthenticationServiceHolder
-                                    .getInstance().getOrganizationUserResidentResolverService()
-                                    .resolveUserFromResidentOrganization(user.getUserName(), null, organizationId);
```

Comment:
Similar work found - https://github.com/wso2-extensions/identity-organization-management-core/pull/15/files
For the federated users, userId set to null as I remember.

Reply:
Other place: https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/blob/15a3b27bd280e6758e21ca2627fc352f0c2f9423/components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java#L107